agent/tools/youcom.py

```
@@ -0,0 +1,199 @@
+
+                results = _merge_sections(data.get("results") or {}, self._param.top_n)
+                self._retrieve_chunks(
+                    results,
+                    get_title=lambda r: r.get("title"),
+                    get_url=lambda r: r.get("url"),
+                    get_content=_result_content,
+                )
+                self.set_output("json", results)
+                logging.info(f"YouComSearch: returned {len(results)} results")
+                return self.output("formalized_content")
+            except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)